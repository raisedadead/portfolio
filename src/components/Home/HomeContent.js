import React from 'react';
import { ClassNames } from '@emotion/core';

import { StaticQuery, graphql } from 'gatsby';
import Box from 'mineral-ui/Box';
import Text from 'mineral-ui/Text';
import Flex from 'mineral-ui/Flex';
import Tooltip from 'mineral-ui/Tooltip';
import Img from 'gatsby-image';

import LinkItem from '../Link';

const Link = props => (
  <ClassNames>
    {({ css }) => (
      <LinkItem
        className={css`
          outline: none;
          text-decoration: none;
          padding: 2px 1px 0;
          border-bottom-style: dashed;
          border-bottom-color: currentcolor;
          border-bottom-width: 1px;
          color: currentcolor;
        `}
        {...props}
      />
    )}
  </ClassNames>
);

export const PureHomeContent = ({ data }) => (
  <Box
    as="section"
    marginHorizontal="auto"
    breakpoints={[`narrow`, `medium`, `wide`]}
    width={[`80vw`, `80vw`, `95vw`, `50vw`]}
  >
    <Flex
      alignItems="center"
      justifyContent="center"
      direction="column"
      height="75vh"
    >
      <Box
        breakpoints={[`narrow`, `medium`, `wide`]}
        width={[`25vw`, `25vw`, `15vw`, `15vw`]}
        margin="1em"
      >
        <Img sizes={data.profileImage.childImageSharp.fluid} />
      </Box>
      <Text as="h2" align="center">
        <Tooltip cursor="help" content="/mru:dʒi:eʃ/" placement="top" usePortal>
          mrugesh
        </Tooltip>
        {` `}
        mohapatra
      </Text>

      <Text as="h3" align="center">
        developer. music addict. open source enthusiast. noob photographer.
      </Text>
      <br />
      <Text as="p" align="center">
        Lead Developer and Community Advocate at{`  `}
        <Link to="https://www.freecodecamp.org" aria-label="freecodecamp.org">
          freeCodeCamp.org
        </Link>
      </Text>
    </Flex>
  </Box>
);

export const HomeContent = () => (
  <StaticQuery
    query={graphql`
      query {
        profileImage: file(relativePath: { eq: "profile.png" }) {
          childImageSharp {
            fluid(maxWidth: 1000) {
              ...GatsbyImageSharpFluid_withWebp_tracedSVG
            }
          }
        }
      }
    `}
    render={data => <PureHomeContent data={data} />}
  />
);

export default HomeContent;
