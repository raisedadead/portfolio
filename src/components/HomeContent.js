import React from 'react';
import { StaticQuery, graphql } from 'gatsby';
import Box from 'mineral-ui/Box';
import Text from 'mineral-ui/Text';
import Flex from 'mineral-ui/Flex';
import Img from 'gatsby-image';

export const PureHomeContent = ({ data }) => (
  <Box
    as="section"
    marginHorizontal="auto"
    breakpoints={[`narrow`, `medium`, `wide`]}
    width={[`80vw`, `80vw`, `95vw`, `50vw`]}
  >
    <Flex
      alignItems="center"
      justifyContent="center"
      direction="column"
      height="75vh"
    >
      <Box
        breakpoints={[`narrow`, `medium`, `wide`]}
        width={[`25vw`, `25vw`, `15vw`, `15vw`]}
        margin="1em"
      >
        <Img sizes={data.profileImage.childImageSharp.fluid} />
      </Box>
      <Text as="h1" align="center">
        mrugesh mohapatra
      </Text>
      <Text as="h2" align="center">
        developer. music addict. open source enthusiast. noob photographer.
      </Text>
      <Text as="h4" align="center">
        Developer Advocate at freeCodeCamp.org
      </Text>
    </Flex>
  </Box>
);

export const HomeContent = () => (
  <StaticQuery
    query={graphql`
      query {
        profileImage: file(relativePath: { eq: "profile.png" }) {
          childImageSharp {
            fluid(maxWidth: 1000) {
              ...GatsbyImageSharpFluid_withWebp_tracedSVG
            }
          }
        }
      }
    `}
    render={data => <PureHomeContent data={data} />}
  />
);

export default HomeContent;
