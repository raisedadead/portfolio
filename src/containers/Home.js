import React, { Component } from 'react'
import {
  // MainNav,
  Footer,
  Background,
  StylesHelper,
  ImageLoader
} from '../components'

import HomeContent from './HomeContent'

const mainContainer = StylesHelper(
  'uk-container',
  'uk-cover-container',
  'uk-width-viewport',
  'uk-height-viewport',
  'uk-padding-remove'
)
const content = StylesHelper('uk-position-center', 'uk-text-center')

const contentSyles = {
  width: '60vw',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center'
}
export default class Home extends Component {
  render() {
    return (
      <div className={mainContainer}>
        <Background data-uk-cover />
        <div
          className="uk-position-cover"
          style={{
            background: 'linear-gradient(to bottom, #36d1dc88, #5b86e5)'
          }}
        />
        <div style={contentSyles} className={content}>
          <div
            style={{ width: '15vw', height: '15vw', borderRadius: '50%' }}
            className="uk-box-shadow-large"
          >
            <ImageLoader src="assets/images/profile.png" alt="profile image" />
          </div>
          <HomeContent />
          {/*
          <MainNav />
          */}
        </div>
        <Footer />
      </div>
    )
  }
}
