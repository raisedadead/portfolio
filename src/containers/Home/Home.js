import React, { Component } from 'react'
import { TopNav, SocialNav, Footer, Background } from '../../components'
import HomeContent from './HomeContent'
import { StylesHelper } from '../../components/'

const mainContainer = StylesHelper(
  'uk-container',
  'uk-cover-container',
  'uk-width-viewport',
  'uk-height-viewport',
  'uk-padding-remove'
)
const content = StylesHelper('uk-position-center', 'uk-text-center')

export default class Home extends Component {
  render() {
    return (
      <div className={mainContainer}>
        <Background data-uk-cover />
        <TopNav />
        <div className={content}>
          <HomeContent />
          <SocialNav />
        </div>
        <Footer />
      </div>
    )
  }
}
