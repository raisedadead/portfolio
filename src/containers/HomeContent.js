import React from 'react'
import { StylesHelper } from '../components/'

const heading = StylesHelper('uk-heading-primary')
const tagline = StylesHelper('uk-text-large')
const content = StylesHelper('uk-width-1-3')
export default props => (
  <div {...props}>
    <img src="assets/images/profile.png" alt="profile" className={content} />
    <p
      className={heading}
      style={{ color: '#F5F5F5', fontFamily: "'Lora', sans-serif" }}
    >
      mrugesh mohapatra
    </p>
    <p
      className={tagline}
      style={{ color: '#D5F7FF', fontFamily: "'Montserrat', sans-serif" }}
    >
      developer. music addict. open source enthusiast. noob photographer.
    </p>
  </div>
)
