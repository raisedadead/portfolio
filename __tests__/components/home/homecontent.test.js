import React from 'react';
import { render } from '@testing-library/react';

import { PureHomeContent as HomeContent } from '../../../src/components/Home/HomeContent';

describe(`HomeContent`, () => {
  it(`renders correctly`, () => {
    const data = {
      profileImage: {
        childImageSharp: {
          fluid: {
            tracedSVG: `random-2b80985b4264e4cb34b1850de91c6911`,
            aspectRatio: 1,
            src: `random-2b80985b4264e4cb34b1850de91c6911`,
            srcSet: `random-2b80985b4264e4cb34b1850de91c6911`,
            srcWebp: `random-2b80985b4264e4cb34b1850de91c6911`,
            srcSetWebp: `random-2b80985b4264e4cb34b1850de91c6911`,
            sizes: `(max-width: 300px) 100vw, 300px`
          }
        }
      }
    };
    const {
      container: { firstChild }
    } = render(<HomeContent data={data} />);
    expect(firstChild).toMatchSnapshot();
  });
});
