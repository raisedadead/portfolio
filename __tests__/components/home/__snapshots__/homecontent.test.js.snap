// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContent renders correctly 1`] = `
<section
  class="css-1hm9igd elghiqq0"
>
  <div
    class="e7r6m60 css-13v9bvk elghiqq0"
  >
    <div
      class="css-1nsbwhm elghiqq0"
    >
      <div
        class="css-10bxytm-Card e1roqw6v0"
      >
        <div
          class=" gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            style="width: 100%; padding-bottom: 100%;"
          />
          <img
            alt=""
            src="random-2b80985b4264e4cb34b1850de91c6911"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
          />
          <picture>
            <source
              sizes="(max-width: 300px) 100vw, 300px"
              srcset="random-2b80985b4264e4cb34b1850de91c6911"
              type="image/webp"
            />
            <img
              alt=""
              sizes="(max-width: 300px) 100vw, 300px"
              src="random-2b80985b4264e4cb34b1850de91c6911"
              srcset="random-2b80985b4264e4cb34b1850de91c6911"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source type='image/webp' srcset="random-2b80985b4264e4cb34b1850de91c6911" sizes="(max-width: 300px) 100vw, 300px" /&gt;&lt;img sizes="(max-width: 300px) 100vw, 300px" srcset="random-2b80985b4264e4cb34b1850de91c6911" src="random-2b80985b4264e4cb34b1850de91c6911" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
      </div>
    </div>
    <h2
      class="css-pdf1eb e19le4zx0"
    >
      <span
        class="css-1h0pl34 ew65k9p0"
        id="tooltip-1"
      >
        <span
          class="css-1wyum7k ew65k9p3"
        >
          <span
            aria-describedby="tooltip-1-content"
            aria-owns="tooltip-1-content"
            class="css-1myw1nk e3ztgjb0"
            role="button"
            tabindex="0"
          >
            mrugesh
          </span>
        </span>
      </span>
       
      mohapatra
    </h2>
    <h3
      class="css-pdf1eb e19le4zx0"
    >
      developer. music addict. open source enthusiast. noob photographer.
    </h3>
    <br />
    <p
      class="css-pdf1eb e19le4zx0"
    >
      Lead Developer and Community Advocate at
        
      <a
        aria-label="freecodecamp.org"
        class="css-3r9jsu"
        href="https://www.freecodecamp.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        freeCodeCamp.org
      </a>
    </p>
  </div>
</section>
`;
